x-pack/plugins/security_solution/server/search_strategy/timeline/factory/events/all/helpers.ts

```diff
@@ -99,22 +97,49 @@ const getValuesFromFields = async (
   );
 };
 
+const buildObjectRecursive = (fieldPath: string, fields: Fields): Partial<Ecs> => {
+  const nestedParentPath = getNestedParentPath(fieldPath, fields);
+  if (!nestedParentPath) {
+    return set({}, fieldPath, toStringArray(get(fieldPath, fields)));
+  }
+
+  const subPath = fieldPath.replace(`${nestedParentPath}.`, '');
+  const subFields = (get(nestedParentPath, fields) ?? []) as Fields[];
+  return set(
+    {},
+    nestedParentPath,
+    subFields.map((subField) => buildObjectRecursive(subPath, subField))
+  );
+};
+
+export const buildObjectForFieldPath = (fieldPath: string, hit: EventHit): Partial<Ecs> => {
+  if (has(fieldPath, hit._source)) {
+    const value = get(fieldPath, hit._source);
+    return set({}, fieldPath, toStringArray(value));
+  }
+
+  return buildObjectRecursive(fieldPath, hit.fields);
```

Comment:
@FrankHassanabad would you be able to pair on verifying the performance of this logic? I know this is a hot path and that we're doing `setTimeout`s and `Promise.resolve`s elsewhere; I would appreciate a crash course in analzying/benchmarking/profiling this!

Reply:
@rylnd and myself discussed this and the conclusion is:

I suggested for testing we can create some large index mappings and testing these functions manually to see if they need to do partitioning or not with a setTimeout.


We have had weird things in the past where once you add one of those, team members from anywhere on any time can sometimes think it's a mistake (no matter how many docs and all CAPS! you put above it :-) ) and they will remove 'em and you have to watch them forever.

But also so we can see that everything functions as we expect, but really right now we don't have an automated perf regression framework for injecting large index mappings, running a test against them and then seeing if we have de-graded performance of the regression of either NodeJS or the frontend UI.

Reply:
There's more context about this in the __Performance Characteristics__ section in the PR description 👍

Reply:
I have PR who removed all the `set` from lodash in this function, if I remember correctly that show slowness in the past.